### ISSUE TITLE
MemoryPack support

### ISSUE THREAD

**Issue by SteveDunn** [SteveDunn/Vogen#696]
I'd love MemoryPack support to be added to Vogen.

_Originally posted by @aradalvand in https://github.com/SteveDunn/Vogen/issues/305#issuecomment-2452321670_

**Comment by SteveDunn** [comment]
@aradalvand - I started working on this, but hit a brick wall: MemoryPack uses source generators, but so does Vogen. There is no way to order which source generator runs first, so MP was emitting errors because value objects were empty (i.e. they hadn't been augmented with fields/constructors etc.)

This won't be a problem if the value objects are in a separate assembly.

**Comment by aradalvand** [comment]
@SteveDunn You're right; I realized the same thing quickly after I suggested this :)
However, it would still have been possible for Vogen to support MemoryPack via generating custom formatters, if MemoryPack had support for this [kind of thing](https://github.com/Cysharp/MemoryPack/issues/347).

**Comment by aradalvand** [comment]
> This won't be a problem if the value objects are in a separate assembly.

Then we'd still be back to square one because the value objects would be considered "foreign" types that the user would have to manually write a formatter for — see [this other issue](https://github.com/Cysharp/MemoryPack/issues/346) I created in MemoryPack's repo.

**Comment by aradalvand** [comment]
The MessagePack is good enough for now; thanks for adding it.

**Comment by SteveDunn** [comment]
👍👍 I'll add the formatters as it'll still be useful for value objects in other assemblies.
